_storage/hello-hub-choice.md

```diff
@@ -0,0 +1,262 @@
+
+4. Leave your new application running and move onto the next section.
+
+## Task 3: Enable hub selection
+
+By default, the app generator assumes you want to use the default flow `redirectToSignIn()` method.  In this section, you replace that method and use the `makeAuthRequest()` method instead. The `makeAuthRequest()` method takes the following parameters:
+
+<dl class="uk-description-list">
+   <dt class="uk-text-lowercase">
+      <code>transitPrivateKey(String = generateAndStoreTransitKey())</code>
+   </dt>
+   <dd>A HEX encoded transit private key.</dd>
+   <dt class="uk-text-lowercase">
+      <code>redirectURI(String = `${window.location.origin}/`)</code>
+   </dt>
+   <dd>Location to redirect the user to after sign in approval.</dd>
+   <dt class="uk-text-lowercase">
+      <code>manifestURI(String = `${window.location.origin}/manifest.json`)</code>
+   </dt>
+   <dd>
+      Location of this app's manifest file.
+      </dd>
+   <dt class="uk-text-lowercase">
+      <code>scopes (Array = DEFAULT_SCOPE)</code>
+   </dt>
+   <dd>The permissions this app is requesting.</dd>
+   <dt class="uk-text-lowercase">
+      <code>appDomain(String = window.location.origin)</code>
+   </dt>
+   <dd>The origin of this app.</dd>
+   <dt class="uk-text-lowercase">
+      <code>expiresAt(Number = nextHour().getTime())</code>
+   </dt>
+   <dd>The time at which this request is no longer valid.</dd>
+   <dt class="uk-text-lowercase">
+      <code>extraParams(Object = {})</code>
+   </dt>
+   <dd>Any extra parameters to pass to the authenticator. Use this to pass options that aren't part of the Blockstack authentication specification, but might be supported by special authenticators.</dd>
+</dl>
+
+To replace the default login, do the following:
+
+1. Using your favorite editor, open the `public/app.js` file.
+2. Locate the `redirectToSignIn()` method at line 4.
+3. Replace `redirectToSignIn()` method with the `blockstack.redirectToSignInWithAuthRequest(authRequest)` method.
+
+   The `authRequest` is the authentication request generated by `makeAuthRequest()` method.
+
+4. Immediately above the method you just added and below the `event.preventDefault()` method, construct a String `const` for the `authRequest`:
+
+    ```
+    const authRequest = blockstack.makeAuthRequest(
+      blockstack.generateAndStoreTransitKey(),
+      'http://localhost:8080/',
+      'http://localhost:8080/manifest.json',
+      ['store_write', 'publish_data'],
+      'http://localhost:8080/',
+      blockstack.nextHour().getTime(), {
+        solicitGaiaHubUrl: true
+      } // new options param
+    );
+    ```
+
+    {% include note.html content="If your app is running a different port than <code>8080</code>, enter that port instead. " %}
+
+   The extra `solicitGaiaHubUrl` parameter of `true` will cause the Blockstack Browser to prompt new identity creators for a storage hub URL.
+
+5. Save and close the `public/app.js` file.
+6. Make sure your app rebuilds cleanly.
```

Comment:
I made the above changes and ran `npm start` manually again, but I'm not encountering the new UI per testing below.

This is what my app.js shows: https://gist.github.com/markmhx/098132dc82ee2927a86e111d1fcd92b8

Reply:
Same (non-) result when I ensure the authRequest constant is within the event listener: https://gist.github.com/markmhx/fb97fadc08425e31aed8eb1ee39ade29

Reply:
Hm strange. I had reset my browser and gone through the create ID flow 🤔

Reply:
Resetting is covered in Step 4.

>     If you are already authenticated with an existing ID, you can choose that ID
   or **Deny**. Choosing an existing ID, signs you into the Hello Blockstack
   application, without offering a hub choice. Instead, choose **Deny**. This
   ends the authentication process. Then, reset the Blockstack Browser.